compiler/src/main/java/com/squareup/anvil/compiler/codegen/KspContributesSubcomponentHandlerSymbolProcessor.kt

```diff
@@ -405,9 +405,18 @@ internal class KspContributesSubcomponentHandlerSymbolProcessor(
               } else {
                 function.asMemberOf(implementingType).returnTypeOrNull()
               }
-            returnTypeToCheck
-              ?.resolveKSClassDeclaration()
-              ?.toClassName() == contributionClassName
+
+            val returnType = returnTypeToCheck?.resolveKSClassDeclaration()
+            if (returnType?.toClassName() == contributionClassName) {
+              return@filter true
+            }
+
+            val isReturningSuperType = returnType != null && contribution.clazz.superTypes.any {
+              it.resolve().resolveKSClassDeclaration() == returnType
+            }
```

Comment:
There is a Resolver API for checking if one type is a subtype of another. I don't remember the name exactly but we should use that one

Reply:
I found `KSType.isAssignableFrom`

So something like this:

```kotlin
returnTypeToCheck.isAssignableFrom(contribution.clazz.asType(emptyList()))
```